config/jsdoc/api/template/static/scripts/main.js

```diff
@@ -236,8 +236,8 @@ $(function () {
     const show = !clsItem.hasClass('toggle-manual-show');
     search.manualToggle(clsItem, show);
   });
-
-  // Auto resizing on navigation
```

Comment:
i dont understand why - 133 ??  the result should be the height of the window ?

Reply:
![openlayers-api-height](https://user-images.githubusercontent.com/4426597/87855283-8588fe80-c917-11ea-8ffc-ca504b543814.png)
And the $el is actually too tall, should be this:
```js
  // Auto resizing on navigation
  var _onResize = function () {
    var height = $(window).height();
    var $el = $('.navigation');
    $el.height(height - 50).find('.navigation-list').height(height - 133);
  };
```
But I'm sure this could be better solved with just css.

Reply:
If you're able to solve this with pure css, please go ahead! That would be much nicer.

Reply:
IMHO the navigation column should be no more hight than the content of the main column (this avoid to have the navigation column and nothing on the right ...)... but sometimes could be higher than window height ... (why not use the space when we have to scroll to see the main column)
not sure than  i can do this in css ... i have to investigate this ...

if i understand actually is not the case and is never higher than the window height minus some margin ..is it right ?
so the js will be better to use the height of the content of the main column ?

Reply:
so i solved this by js (in pure css i dont see how to get and compare the height of differents element ? ...)

```
//  Auto resizing on navigation
  var _onResize = function () {
    var height_w = $(window).height();
    var height_s = $('section').height();
    var $el = $('.navigation');
    var dif_h = height_w - height_s;
    if (window.matchMedia("(min-width: 768px)").matches) {
      if (dif_h >=0){
         $('.navigation').height(height_s+dif_h+74);
         $('.navigation-list').height(height_s+dif_h)
      }
      else {
         $('.navigation').height(height_s+74);
         $('.navigation-list').height(height_s-60);
      }
    }
    else{
      $('.navigation').height(220);
      $('.navigation-list').height('100%');
    }
  }
  $(window).on('resize', _onResize);
  _onResize();
```

Reply:
So I've been trying to do this with css only. It is seems to work with the css `calc()` function if the header and search bar have a fixed size (50px each in the example page below)
Not sure if it is worth changing it, maybe at a later time  ...

```html
<!DOCTYPE html>
<html>
  <head>
    <title>Test</title>
    <style>
      * {
        box-sizing: border-box;
      }
      html, body {
        margin: 0;
        height: 100vh;
      }
      body {
        display: flex;
        flex-flow: column;
      }
      .navbar {
        background-color: #1F6B75;
        height: 50px;
      }
      .wrap {
        display: flex;
        flex-direction: row;
      }
      .navigation {
        background-color: #333;
        color: #ddd;
        height: calc(100vh - 50px);

        flex-basis: 250px;
        flex-grow: 0;
        flex-shrink: 0;
        display: flex;
        flex-flow: column;
      }
      .navigation > .search {
        padding: 10px;
      }
      .navigation > .search > input {
        width: 100%;
        font-size: 12px;
        line-height: 1.5;
        height: 30px;
        border-radius: 3px;
        padding: 5px 10px;
      }
      .navigation-list {
        overflow: auto;
        background-color: #444;
        height: calc(100vh - 50px - 50px - 23px);
      }
      .navigation ul {
        margin: 0;
        padding-left: 10px;
      }
      .navigation li {
        list-style: none;
        padding-left: 0px;
        padding-top: 8px;
      }
      .main {
        overflow: auto;
        padding: 10px;
        height: calc(100vh - 50px);
      }
      @media (max-width: 768px) {
        .wrap {
          flex-direction: column;
          height: calc(100vh - 50px);
          overflow: auto;
        }
        .navigation {
          max-height: 250px;
          flex-basis: 50px;
          height: initial;
        }
        .navigation-list {
          max-height: calc(250px - 50px);
          height:  inherit;
        }
        .main {
          height: inherit;
          overflow: visible;
        }
      }
    </style>
  </head>
  <body>
    <div class="navbar"></div>
    <div class="wrap">
      <div class=navigation>
        <div class="search">
          <input type="text">
        </div>
        <ul class="navigation-list">
          <li>Test 1 Test 1 Test 1 Test 1 Test 1 Test 1 </li>
          <li>Test 2</li>
          <li>Test 3</li>
          <li>Test 4</li>
          <li>Test 5</li>
          <li>Test 6</li>
          <li>Test 7</li>
          <li>Test 8</li>
          <li>Test 9</li>
          <li>Test 10</li>
          <li>Test 11</li>
          <li>Test 12</li>
          <li>Test 13</li>
          <li>Test 14</li>
          <li>Test 15</li>
          <li>Test 16</li>
          <li>Test 17</li>
          <li>Test 18</li>
          <li>Test 19</li>
          <li>Test 20</li>
          <li>Test 21</li>
          <li>Test 22</li>
          <li>Test 23</li>
          <li>Test 24</li>
          <li>Test 25</li>
          <li>Test 26</li>
          <li>Test 27</li>
          <li>Test 28</li>
          <li>Test 29</li>
          <li>Test 30</li>
          <li>Test 31</li>
          <li>Test 32</li>
          <li>Test 33</li>
          <li>Test 34</li>
          <li>Test 35</li>
          <li>Test 36</li>
          <li>Test 37</li>
          <li>Test 38</li>
          <li>Test 39</li>
          <li>Test 40</li>
          <li>Test 41</li>
          <li>Test 42</li>
          <li>Test 43</li>
          <li>Test 44</li>
          <li>Test 45</li>
          <li>Test 46</li>
          <li>Test 47</li>
          <li>Test 48</li>
          <li>Test 49</li>
          <li>Test 50</li>
        </ul>
      </div>
      <div class="main">
        <h1>Lorem Ipsum</h1>
        <h2>"Neque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit..."</h2>
        <h2>"There is no one who loves pain itself, who seeks after it and wants to have it, simply because it is pain..."</h2>
        <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam at nisi quis urna gravida pretium. Praesent id blandit sapien. Vestibulum blandit tincidunt fringilla. Nulla tristique massa sit amet tortor vestibulum ornare. In tristique lectus vitae nunc venenatis posuere. Suspendisse elementum urna ac purus convallis, non imperdiet neque scelerisque. Curabitur commodo id enim in porttitor. Donec imperdiet bibendum porta. Vivamus sodales, libero ac condimentum interdum, odio neque porttitor elit, quis dignissim metus tortor sit amet arcu. Integer vestibulum nulla eu nisi commodo, ac gravida felis vehicula. Nullam nisi justo, auctor sit amet viverra nec, malesuada ut purus. Integer nec quam placerat, egestas nunc id, rhoncus justo. Nunc aliquam turpis nibh, ut rhoncus neque laoreet nec. Pellentesque dapibus nibh at vehicula accumsan.</p>
        <p>Sed lorem neque, malesuada quis ullamcorper eu, euismod sit amet turpis. Nunc fringilla vestibulum nunc, nec iaculis mi fringilla nec. Aenean feugiat, metus eu finibus imperdiet, nunc enim cursus diam, in vehicula ante purus at leo. Etiam et lobortis elit. Vivamus suscipit nunc at libero rhoncus interdum. Mauris tempor euismod nulla id auctor. Integer placerat mollis felis, in tincidunt ipsum vulputate vel. Suspendisse ultricies, justo at cursus facilisis, dolor augue viverra sapien, sit amet scelerisque lacus tellus ac tellus. Aliquam euismod nunc sit amet ligula maximus, quis viverra tellus tincidunt. Etiam purus eros, mattis eu mollis nec, facilisis ut eros. Ut dapibus, lacus sed blandit pretium, nunc ligula pellentesque sem, vitae commodo nunc metus non orci.</p>
        <p>Nulla fringilla maximus nulla id gravida. Proin nec ultrices sapien. Duis mollis nec felis placerat scelerisque. Morbi a volutpat massa. Duis at sodales nulla. Donec eget dui quis nisi ullamcorper ornare et ac metus. Pellentesque libero tellus, tristique non lectus eu, tempor eleifend mauris. Nam aliquet lacus sed augue sagittis scelerisque. Etiam ultricies commodo augue. Nam sed hendrerit sapien. In molestie urna eu urna cursus, a suscipit massa bibendum. In sodales aliquam orci in venenatis. Vivamus rhoncus sem fermentum odio malesuada, ut tempus arcu ultrices. Aenean viverra sapien a pulvinar porta. Quisque ac egestas est, non tempus ante. Integer fermentum eget magna nec tincidunt.</p>
        <p>Ut id velit a est cursus hendrerit. Donec lobortis maximus ex, ac suscipit purus accumsan eget. Duis commodo placerat libero. Vestibulum erat tellus, pharetra vel sagittis ut, euismod id risus. In in feugiat sapien, sit amet semper tortor. Etiam imperdiet molestie diam id luctus. Cras pulvinar tellus at arcu lacinia, ac varius leo pharetra.</p>
        <p>Nunc molestie imperdiet odio, id accumsan erat varius et. Maecenas a maximus ligula, vitae consectetur sem. In quis dui nibh. Aliquam nulla sapien, aliquet eget metus a, rutrum maximus urna. Mauris euismod hendrerit pulvinar. Sed nisl lectus, pharetra ut tempor id, mollis laoreet nisi. Nunc vel ex nisi. Mauris porta nulla suscipit orci tristique, et mattis orci finibus. Nullam vel augue magna. Nullam euismod rhoncus felis, sit amet pretium dolor porta nec. </p>
      </div>
    </div>
  </body>
</html>
```

Reply:
@tschaub thanks i will try this in my next investigation  ! why i not think to vh ... but sure in css it will be better way